library/src/main/java/com/novoda/downloadmanager/lib/Request.java

```diff
@@ -234,6 +235,17 @@ public Request alwaysAttemptResume() {
     }
 
     /**
+     * Automatically pause the download when reaching the end of the file instead of writing the last bytes to disk.
+     * This allows for a resume of the download against an updated tar file.
+     * If used this also sets {@code alwaysAttemptResume()}
+     */
+    public Request allowTarUpdates() {
+        allowTarUpdates = true;
+        alwaysAttemptResume();
```

Comment:
@amlcurran I had a chat with @zegnus he is working on a flag to enforce the no_intergrity.
This PR can probably be merged as is and @zegnus can then modify the Request builder to enforce the `no_intergrity` instead of `alwaysAttemptResume` in the future.